src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -291,9 +292,17 @@ async def validate_rsync_task(self, data, schema):
                                 errors='ignore',
                             )).stdout
 
-                            await self.middleware.run_in_thread(known_hosts_path.write_text, known_hosts_text)
-                            await self.middleware.run_in_thread(os.chown, known_hosts_path, user['pw_uid'],
-                                                                user['pw_gid'])
+                            # If for whatever reason the dir does not exist, let's create it
+                            # An example of this is when we run rsync tests we nuke the directory
+                            def handle_ssh_dir():
+                                with contextlib.suppress(FileExistsError):
+                                    ssh_dir_path.mkdir(0o700)
+                                    os.chown(ssh_dir_path.absolute(), user['pw_uid'], user['pw_gid'])
+
+                                known_hosts_path.write_text(known_hosts_text)
+                                os.chown(known_hosts_path.absolute(), user['pw_uid'], user['pw_gid'])
+
+                            await self.middleware.run_in_thread(handle_ssh_dir)
```

Comment:
It was actually not necessary to complicate things here, `run_in_thread` reuses threads

Reply:
"complicate" is relative here. `run_in_thread` reusing threads implies the developer has a deep understanding of how our threadpool operates. What I suggested, instead, clearly shows that a single function will be called in a single thread.